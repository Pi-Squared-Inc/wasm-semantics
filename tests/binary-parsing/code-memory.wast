(module $mymodule
  (func (param i32) (result i32)
    i32.const 0
    i32.load
    drop

    i32.const 0
    i32.load offset=8
    drop

    i32.const 0
    i32.load align=4
    drop

    i32.const 0
    i32.load offset=8 align=4
    drop

    i32.const 0
    i64.load offset=8
    drop

    i32.const 0
    f32.load offset=8
    drop

    i32.const 0
    f64.load offset=8
    drop

    i32.const 0
    i32.load8_s offset=8
    drop

    i32.const 0
    i32.load8_u offset=8
    drop

    i32.const 0
    i32.load16_s offset=8
    drop

    i32.const 0
    i32.load16_u offset=8
    drop

    i32.const 0
    i64.load8_s offset=8
    drop

    i32.const 0
    i64.load8_u offset=8
    drop

    i32.const 0
    i64.load16_s offset=8
    drop

    i32.const 0
    i64.load16_u offset=8
    drop

    i32.const 0
    i64.load32_s offset=8
    drop

    i32.const 0
    i64.load32_u offset=8
    drop

    i32.const 0
    i32.const 0
    i32.store offset=8

    i32.const 0
    i32.const 0
    i32.store8 offset=8

    i32.const 0
    i32.const 0
    i32.store16 offset=8

    i32.const 0
    i64.const 0
    i64.store offset=8

    i32.const 0
    i64.const 0
    i64.store8 offset=8

    i32.const 0
    i64.const 0
    i64.store16 offset=8

    i32.const 0
    i64.const 0
    i64.store32 offset=8

    i32.const 0
    i32.const 0
    f32.load offset=8
    f32.store offset=8

    i32.const 0
    i32.const 0
    f64.load offset=8
    f64.store offset=8

    memory.size
    drop

    i32.const 0
    memory.grow
    drop

    i32.const 0
    i32.const 0
    i32.const 0
    memory.fill

    i32.const 0
    i32.const 0
    i32.const 0
    memory.copy

    i32.const 0
    i32.const 0
    i32.const 0
    memory.init 0

    data.drop 0
    (return (i32.const 1))
  )
  (memory (;0;) 17)
  (global $g (mut i32) (i32.const 1048576))
  (data (i32.const 1048576) ";askdfja;skdjf")
)
