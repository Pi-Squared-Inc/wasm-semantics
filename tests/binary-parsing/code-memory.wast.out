<generatedTop>
  <k>
    #module (... types: #type (... type: [ i32  .ValTypes ] -> [ i32  .ValTypes ] , metadata:  )  .EmptyStmts , funcs: #func (... type: 0 , locals: [ .ValTypes ] , body: i32 . const 0  #load ( i32 , load , 0 )  drop  i32 . const 0  #load ( i32 , load , 8 )  drop  i32 . const 0  #load ( i32 , load , 0 )  drop  i32 . const 0  #load ( i32 , load , 8 )  drop  i32 . const 0  #load ( i64 , load , 8 )  drop  i32 . const 0  #load ( f32 , load , 8 )  drop  i32 . const 0  #load ( f64 , load , 8 )  drop  i32 . const 0  #load ( i32 , load8_s , 8 )  drop  i32 . const 0  #load ( i32 , load8_u , 8 )  drop  i32 . const 0  #load ( i32 , load16_s , 8 )  drop  i32 . const 0  #load ( i32 , load16_u , 8 )  drop  i32 . const 0  #load ( i64 , load8_s , 8 )  drop  i32 . const 0  #load ( i64 , load8_u , 8 )  drop  i32 . const 0  #load ( i64 , load16_s , 8 )  drop  i32 . const 0  #load ( i64 , load16_u , 8 )  drop  i32 . const 0  #load ( i64 , load32_s , 8 )  drop  i32 . const 0  #load ( i64 , load32_u , 8 )  drop  i32 . const 0  i32 . const 0  #store ( i32 , store , 8 )  i32 . const 0  i32 . const 0  #store ( i32 , store8 , 8 )  i32 . const 0  i32 . const 0  #store ( i32 , store16 , 8 )  i32 . const 0  i64 . const 0  #store ( i64 , store , 8 )  i32 . const 0  i64 . const 0  #store ( i64 , store8 , 8 )  i32 . const 0  i64 . const 0  #store ( i64 , store16 , 8 )  i32 . const 0  i64 . const 0  #store ( i64 , store32 , 8 )  i32 . const 0  i32 . const 0  #load ( f32 , load , 8 )  #store ( f32 , store , 8 )  i32 . const 0  i32 . const 0  #load ( f64 , load , 8 )  #store ( f64 , store , 8 )  memory.size  drop  i32 . const 0  memory.grow  drop  i32 . const 0  i32 . const 0  i32 . const 0  memory.fill  i32 . const 0  i32 . const 0  i32 . const 0  memory.copy  i32 . const 1  return  .EmptyStmts , metadata: #meta (... id:  , localIds: .Map ) )  .EmptyStmts , tables: .EmptyStmts , mems: #memory (... limits: #limitsMin ( 17 ) , metadata:  )  .EmptyStmts , globals: #global (... type: var i32 , init: i32 . const 1048576  .EmptyStmts , metadata:  )  .EmptyStmts , elem: .EmptyStmts , data: #data (... index: 0 , offset: i32 . const 1048576  .EmptyStmts , data: b";askdfja;skdjf" )  .EmptyStmts , start: .EmptyStmts , importDefns: .EmptyStmts , exports: .EmptyStmts , metadata: #meta (... id:  , funcIds: .Map , filename: .String ) ) ~> .K
  </k>
  <wasm>
    <instrs>
      .K
    </instrs>
    <valstack>
      .ValStack
    </valstack>
    <curFrame>
      <locals>
        .List
      </locals>
      <curModIdx>
        .Int
      </curModIdx>
    </curFrame>
    <moduleRegistry>
      .Map
    </moduleRegistry>
    <moduleIds>
      .Map
    </moduleIds>
    <moduleInstances>
      .ModuleInstCellMap
    </moduleInstances>
    <nextModuleIdx>
      0
    </nextModuleIdx>
    <mainStore>
      <funcs>
        .FuncDefCellMap
      </funcs>
      <nextFuncAddr>
        0
      </nextFuncAddr>
      <tabs>
        .TabInstCellMap
      </tabs>
      <nextTabAddr>
        0
      </nextTabAddr>
      <mems>
        .List
      </mems>
      <globals>
        .GlobalInstCellMap
      </globals>
      <nextGlobAddr>
        0
      </nextGlobAddr>
      <elems>
        .ElemInstCellMap
      </elems>
      <nextElemAddr>
        0
      </nextElemAddr>
    </mainStore>
    <deterministicMemoryGrowth>
      true
    </deterministicMemoryGrowth>
  </wasm>
</generatedTop>
